### PR TITLE
Fspt 177 apply submit to assess

### PR DESCRIPTION
**Pull request by srh-sloan**
### Change description
https://mhclgdigital.atlassian.net/browse/FSPT-177
This PR stops application-store from writing to the import queue for assessment. Instead, it calls assessment-store directly to copy the data into the assessment tables. Covered in this PR:
- Updated `submit_application` to call assessment store functions directly to derive application values and insert the row.
- Moved the `send_submit_notification` code into a separate function for easier readability
- Unit tests for the new submit route and functions
- Updated error handling code so that:
    - Submit will not fail for the user if the notification fails to send, or if we can't derive application values

Out of scope:
- Removing the code to read from the queue in assessment. This de-risks the change when it goes into prod and means no data will be lost if someone submits using the old method as we deploy the change.

- [x] Unit tests and other appropriate tests added or updated
- [ ] README and other documentation has been updated / added (if needed)
- [ ] Commit messages are meaningful and follow good commit message guidelines (e.g. "FS-XXXX: Add margin to nav items preventing overlapping of logo")

### How to test
Pull this branch.
Create and submit an application
You should see it in assessment as normal.
No changes should be visible to the user

### Screenshots of UI changes (if applicable)
N/A

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -16,5 +11,10 @@
       },
       "editor.formatOnSave": true,
       "editor.defaultFormatter": "charliermarsh.ruff"
-    }
+    },
+    "python.testing.pytestArgs": [
```
Comment (gidsg) [inline .vscode/settings.json:15]: These changes are unrelated so could be in a separate PR?

File: application_store/_helpers/application.py [hunk 2]
```diff
@@ -19,3 +33,57 @@ def order_applications(applications, order_by, order_rev):
...
+                application.round_id,
+            ).get("lead_contact_name", "")
+        )
+
+        if Decision(eoi_decision) == Decision.PASS:  # EOI Full pass
+            notify_template = Config.NOTIFY_TEMPLATE_EOI_PASS
+
+        elif Decision(eoi_decision) == Decision.PASS_WITH_CAVEATS:  # EOI Pass with caveats
+            notify_template = Config.NOTIFY_TEMPLATE_EOI_PASS_W_CAVEATS
+            contents = [NotifyConstants.APPLICATION_CAVEATS] = (eoi_results["caveats"],)
```
Comment (gidsg) [inline application_store/_helpers/application.py:71]: Could we refactor this using [Structural Pattern Matching](https://docs.python.org/3/whatsnew/3.10.html#pep-634-structural-pattern-matching)?

File: application_store/_helpers/application.py [hunk 3]
```diff
@@ -19,3 +33,57 @@ def order_applications(applications, order_by, order_rev):
...
+    else:
+        notify_template = Config.NOTIFY_TEMPLATE_SUBMIT_APPLICATION
+        eoi_decision = None
+        full_name = account.full_name
+
+    contents["application"] = create_qa_base64file(contents.get("application"), True)
+    del contents["application"]["forms"]
```
Comment (gidsg) [inline application_store/_helpers/application.py:78]: Rather than mutating here could we return the thing we want directly e.g.
`application_contents = create_qa_base64file(contents.get("application"), True)` and pass that to the `Notification`?

File: assessment_store/db/queries/assessment_records/_helpers.py [hunk 4]
```diff
@@ -10,147 +10,189 @@
...
+        "funding_amount_requested": 0,
+        "asset_type": "No asset type specified.",
+        "language": application_json["language"],
+        "location_json_blob": {
+            "error": False,
+            "postcode": FIELD_DEFAULT_VALUE,
+            "county": FIELD_DEFAULT_VALUE,
+            "region": FIELD_DEFAULT_VALUE,
+            "country": FIELD_DEFAULT_VALUE,
+            "constituency": FIELD_DEFAULT_VALUE,
```
Comment (gidsg) [inline assessment_store/db/queries/assessment_records/_helpers.py:118]: Would prefer for this to be `null` and the frontend handle what to display. But OK if this is a lift and shift of the previous behaviour.

Comment (srh-sloan) [reply assessment_store/db/queries/assessment_records/_helpers.py:118]: Yeah it's just what was there before, didn't want to have to change the frontend as well

File: .vscode/settings.json (hunk 1 above)
Comment (srh-sloan) [reply .vscode/settings.json:15]: This can just go actually - I had to remove and re-add and VS code has just put in a different place in the file.

File: uv.lock [hunk 5]
```diff
@@ -325,6 +325,30 @@ uvicorn = [
     { name = "uvicorn", extra = ["standard"] },
 ]
 
+[[package]]
```
[the remark this answers is not in the document]
Comment (srh-sloan) [reply uv.lock:328]: I added `pytest-cov` when writing the tests so it's brought these in but happy to move to a different PR 👍

File: application_store/db/models/application/applications.py [hunk 6]
```diff
@@ -42,7 +42,11 @@ class Applications(BaseModel):
     __table_args__ = (db.UniqueConstraint("fund_id", "round_id", "key", name="_reference"),)
 
     def as_dict(self):
-        date_submitted = self.date_submitted.isoformat() if self.date_submitted else "null"
+        date_submitted = (
+            (self.date_submitted if type(self.date_submitted) is str else self.date_submitted.isoformat())
```
Comment (samuelhwilliams) [inline application_store/db/models/application/applications.py:46]: this feels odd - I think we should have this always as a datetime instance - do you know where the string version is coming from?

File: application_store/_helpers/application.py (hunk 3 above)
Comment (srh-sloan) [reply application_store/_helpers/application.py:78]: Have tweaked this slightly but reluctant to change any further as it's feeling less like a shift and shift and that function is used in other places.

File: application_store/_helpers/application.py (hunk 2 above)
Comment (srh-sloan) [reply application_store/_helpers/application.py:71]: Yep done

File: application_store/db/models/application/applications.py (hunk 6 above)
Comment (samuelhwilliams) [reply application_store/db/models/application/applications.py:46]: sorry yep ignore me - we still are reading from the queue for one deployment 👍

**Review by samuelhwilliams — DISMISSED** [review]
have tested this locally and it works, so don't want to block too much ✅ 

But there is lots of juicy stuff here to consolidate in the next few weeks/months...

File: application_store/_helpers/application.py [hunk 7]
```diff
@@ -19,3 +33,53 @@ def order_applications(applications, order_by, order_rev):
...
+        match eoi_decision:
+            case Decision.FAIL:
+                return  # we don't send emails for a failure
+
+            case Decision.PASS:
+                notify_template = Config.NOTIFY_TEMPLATE_EOI_PASS
+
+            case Decision.PASS_WITH_CAVEATS:
+                notify_template = Config.NOTIFY_TEMPLATE_EOI_PASS_W_CAVEATS
+                contents[NotifyConstants.APPLICATION_CAVEATS] = eoi_results["caveats"]
```
[the remark this answers is not in the document]
Comment (srh-sloan) [reply application_store/_helpers/application.py:71]: Done

File: application_store/_helpers/application.py [hunk 8]
```diff
@@ -19,3 +33,53 @@ def order_applications(applications, order_by, order_rev):
...
+    account,
+    application_with_form_json_and_fund_name,
+    application,
+    round_data,
+):
+    contents = {
+        NotifyConstants.APPLICATION_FIELD: create_qa_base64file(application_with_form_json_and_fund_name, True),
+        NotifyConstants.MAGIC_LINK_CONTACT_HELP_EMAIL_FIELD: round_data.contact_email,
+    }
+    del contents[NotifyConstants.APPLICATION_FIELD]["forms"]
```
[the remark this answers is not in the document]
Comment (srh-sloan) [reply application_store/_helpers/application.py:50]: Added a unit test. Agree with the worry around `del` - it was already in there and is another thing we can properly refactor after the lift/shift queue removal

**Review by samuelhwilliams — APPROVED** [review]
